Dockerfile-cuda

```diff
@@ -108,6 +108,16 @@ ENV HUGGINGFACE_HUB_CACHE=/data \
     PORT=80 \
     USE_FLASH_ATTENTION=$DEFAULT_USE_FLASH_ATTENTION
 
+# Amazon SageMaker compatible image
```

Comment:
Lets put this also only in all https://github.com/huggingface/text-embeddings-inference/blob/main/Dockerfile-cuda-all

Reply:
Moved `sagemaker` target to `Dockerfile-cuda-all` and updated `sagemaker-entrypoint.sh`. 

Ran into this issue https://github.com/huggingface/candle/issues/733 while running some tests on `g5.2xlarge` endpoints - added a `CUDA_COMPUTE_CAP` env var as a workaround.

![image](https://github.com/huggingface/text-embeddings-inference/assets/7282984/969de419-d33e-4a86-8d6b-a958483870ee)